optimizely/config/polling_manager_test.go

```diff
@@ -74,3 +76,57 @@ func TestNewPollingProjectConfigManagerWithNull(t *testing.T) {
 	_, err := configManager.GetConfig()
 	assert.NotNil(t, err)
 }
+
+func TestNewPollingProjectConfigManagerWithSimilarDatafileRevisions(t *testing.T) {
+	mockDatafile1 := []byte(`{"revision":"42","projectId":"1000"}`)
+	mockDatafile2 := []byte(`{"revision":"42","projectId":"1001"}`)
+	projectConfig1, _ := datafileprojectconfig.NewDatafileProjectConfig(mockDatafile1)
+	mockRequester := new(MockRequester)
+	mockRequester.On("Get", []utils.Header(nil)).Return(mockDatafile1, 200, nil)
+
+	sdkKey := "test_sdk_key"
+	options := PollingProjectConfigManagerOptions{
+		Requester: mockRequester,
+	}
+
+	exeCtx := utils.NewCancelableExecutionCtx()
+	configManager := NewPollingProjectConfigManagerWithOptions(exeCtx, sdkKey, options)
+	mockRequester.AssertExpectations(t)
+
+	actual, err := configManager.GetConfig()
+	assert.Nil(t, err)
+	assert.NotNil(t, actual)
+	assert.Equal(t, projectConfig1, actual)
+
+	configManager.SyncConfig(mockDatafile2)
+	actual, err = configManager.GetConfig()
+	assert.Equal(t, projectConfig1, actual)
+}
+
+func TestNewPollingProjectConfigManagerWithDifferentDatafileRevisions(t *testing.T) {
+	mockDatafile1 := []byte(`{"revision":"42","projectId":"1000"}`)
```

Comment:
Same as above.

Reply:
I have updated the projectId for this test.